### PR TITLE
Add descriptors: max_read_image_args, max_write_image_args

### LINKED ISSUE #317: Add device descriptors: max_read_image_args, max_write_image_args
Part of the task #286

- [ ] info::device::max_read_image_args (uint32_t)
- [ ] info::device::max_write_image_args (uint32_t)

### PR DESCRIPTION
**Pull request by 1e-to**
Closes #317

### CODE REVIEW

**Comment by oleksandr-pavlyk** [comment]
I think my comment on another PR applies here as well. 

On the Python side there is no need to ask for these descriptors at the time of `SyclDevice` construction. The query should be made on as-needed bases, i.e. when user queries the instance property.

This will avoid need to special-handle descriptors in case they are not supported for certain devices, like `max_sub_group_size` may not be supported on host device, and SYCL throws an exception.

File: dpctl-capi/source/dpctl_sycl_device_interface.cpp [hunk 1]
```diff
@@ -388,6 +388,24 @@ bool DPCTLDevice_HasAspect(__dpctl_keep const DPCTLSyclDeviceRef DRef,
     return hasAspect;
 }
 
+#define declmethod(FUNC, NAME)                                                 \
```
Comment (diptorupd) [inline dpctl-capi/source/dpctl_sycl_device_interface.cpp:391]: You are really liking this style :+1:

File: dpctl-capi/tests/test_sycl_device_interface.cpp [hunk 2]
```diff
@@ -275,6 +275,38 @@ TEST_P(TestDPCTLSyclDeviceInterface, Chk_GetPreferredVectorWidthHalf)
     }
 }
 
+TEST_P(TestDPCTLSyclDeviceInterface, Chk_GetMaxReadImageArgs)
+{
+    DPCTLSyclDeviceRef DRef = nullptr;
+    size_t max_read_image_args = 0;
+    EXPECT_NO_FATAL_FAILURE(DRef = DPCTLDevice_CreateFromSelector(DSRef));
+    if (!DRef)
```
Comment (diptorupd) [inline dpctl-capi/tests/test_sycl_device_interface.cpp:283]: See my comments in #315. Same apply here.

**Review by diptorupd — CHANGES_REQUESTED** [review]
Rebase with master and update C API tests.

**Review by oleksandr-pavlyk — APPROVED** [review]

**Comment by diptorupd** [comment]
@1e-to can you fix the merge conflicts. The macros in #315 and #318 should have different names.

**Review by oleksandr-pavlyk — APPROVED** [review]

**Review by diptorupd — APPROVED** [review]

**Pull request merged**